types/datum.go

```diff
@@ -61,7 +62,7 @@ const (
 // It has better performance and is easier to use than `interface{}`.
 type Datum struct {
 	k         byte        // datum kind.
-	collation uint8       // collation can hold uint8 values.
+	collation string      // collation hold the collation information for string value.
```

Comment:
I still prefer using an integer(maybe `uint16`?) for collation, and we can use `GetCollatorByID()` to get them. After all, integer use much more memory than strings.